File: Autolib/ClientApp/src/app/models/vehicule.ts
import { Reservation } from "./reservation";
import { TypeVehicule } from "./type-vehicule";
import { Utiliser } from "./utiliser";

export class Vehicule {
  id: number;
  rfid: number;
  etatBatterie: number;
  disponibilite: string;
  latitude: number;
  longitude: number;
  typeVehicule: TypeVehicule;
  reservations: Reservation[];
  utilisations: Utiliser[];

  public isUsed(used: boolean): void {
    if (used) this.disponibilite = 'UTILISE'
    else this.disponibilite = 'LIBRE';
  }

  public isBooked(booked: boolean): void {
    if (booked) this.disponibilite = 'RESERVEE'
    else this.disponibilite = 'LIBRE'
  }
}

File: Autolib/Vehicule.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace Autolib
{
    public partial class Vehicule
    {
        public Vehicule()
        {
            Bornes = new HashSet<Borne>();
            Reservations = new HashSet<Reservation>();
            Utilises = new HashSet<Utilise>();
        }

        public int IdVehicule { get; set; }
        public int Rfid { get; set; }
        public int? EtatBatterie { get; set; }
        public string Disponibilite { get; set; }
        public decimal? Latitude { get; set; }
        public decimal? Longitude { get; set; }
        public int TypeVehicule { get; set; }

        public virtual TypeVehicule TypeVehiculeNavigation { get; set; }
        public virtual ICollection<Borne> Bornes { get; set; }
        public virtual ICollection<Reservation> Reservations { get; set; }
        public virtual ICollection<Utilise> Utilises { get; set; }
    }
}
